docs/docs/examples/tools/casssandra.ipynb

```diff
@@ -0,0 +1,396 @@
+    "\n",
+    "```json\n",
+    "You are an Apache Cassandra expert query analysis bot with the following features \n",
+    "and rules:\n",
+    " - You will take a question from the end user about finding specific \n",
+    "   data in the database.\n",
+    " - You will examine the schema of the database and create a query path. \n",
+    " - You will provide the user with the correct query to find the data they are looking \n",
+    "   for, showing the steps provided by the query path.\n",
+    " - You will use best practices for querying Apache Cassandra using partition keys \n",
+    "   and clustering columns.\n",
+    " - Avoid using ALLOW FILTERING in the query.\n",
+    " - The goal is to find a query path, so it may take querying other tables to get \n",
+    "   to the final answer. \n",
+    "\n",
+    "The following is an example of a query path in JSON format:\n",
+    "\n",
+    " {\n",
+    "  \"query_paths\": [\n",
+    "    {\n",
+    "      \"description\": \"Direct query to users table using email\",\n",
+    "      \"steps\": [\n",
+    "        {\n",
+    "          \"table\": \"user_credentials\",\n",
+    "          \"query\": \n",
+    "             \"SELECT userid FROM user_credentials WHERE email = 'example@example.com';\"\n",
+    "        },\n",
+    "        {\n",
+    "          \"table\": \"users\",\n",
+    "          \"query\": \"SELECT * FROM users WHERE userid = ?;\"\n",
+    "        }\n",
+    "      ]\n",
+    "    }\n",
+    "  ]\n",
+    "}\n",
+    "```\n",
+    "\n",
+    "## Tools Provided\n",
+    "\n",
+    "### `cassandra_db_schema`\n",
+    "Gathers all schema information for the connected database or a specific schema. Critical for the agent when determining actions. \n",
+    "\n",
+    "### `cassandra_db_select_table_data`\n",
+    "Selects data from a specific keyspace and table. The agent can pass paramaters for a predicate and limits on the number of returned records. \n",
+    "\n",
+    "### `cassandra_db_query`\n",
+    "Expiriemental alternative to `cassandra_db_select_table_data` which takes a query string completely formed by the agent instead of parameters. *Warning*: This can lead to unusual queries that may not be as performant(or even work). This may be removed in future releases. If it does something cool, we want to know about that too. You never know!"
```

Comment:
```suggestion
    "Experimental alternative to `cassandra_db_select_table_data` which takes a query string completely formed by the agent instead of parameters. *Warning*: This can lead to unusual queries that may not be as performant(or even work). This may be removed in future releases. If it does something cool, we want to know about that too. You never know!"
```